stable/nginx/Chart.yaml

```diff
@@ -0,0 +1,15 @@
+---
+name: nginx
+version: 0.1.0
+appVersion: 1.15.12
+home: https://hub.docker.com/_/nginx
+description: A Helm chart for NGINX
+keywords:
+  - nginx
+  - server
+sources:
+  - https://github.com/nginxinc/docker-nginx
+  - https://github.com/nginx/nginx
+maintainers:
+  - name: joejulian
+    email: me@joejulian.name
```

Comment:
I would be in favor of adding this to [helm hub](https://github.com/helm/hub/blob/c23b0911828096e30f88af7b8925b78fd9faae67/Repositories.md#guidelines-for-repository-inclusion) rather than directly to stable (or at a minimum add to incubator first).

There are lots of use cases for nginx, and this makes getting a generic chart right pretty tough. I think having the chance to iterate without the baggage of maintaining stability for a bit would be valuable.

Reply:
@jlegrone afaik we are not adding chart in the incubator anymore.
I'm also in favor to add to the helm hub

Reply:
I'm not against it being added to helm hub too, but we do need this in the default repo. According to the documentation, once I satisfy the review comments this should qualify as being considered "stable".

Reply:
So from past experience maintaining an internal "generic nginx" chart, here are some things I would be looking for in a chart occupying the `stable/nginx` namespace:

1. Support serving static files from arbitrary directory paths and at arbitrary [locations](http://nginx.org/en/docs/http/ngx_http_core_module.html#location)
2. The same as 1) but for "single page apps"
    - This could actually be accomplished through a combination of 1) and 5) but it's a very common use case.
3. Support configuring proxies at arbitrary locations
4. Enable mounting user-defined volumes and use the `tpl` directive so that configmaps, secrets, pvcs, etc defined in a parent chart can be dynamically referenced
5. Allow the injection of user-defined nginx configuration in the `server` block and each `location` block
6. Optionally, support TLS termination
    - This could be added on later and might be possible for users to implement themselves through a combination of 4) and 5).

Reply:
Most of this can be satisfied by the ability to replace `/etc/nginx` with a ConfigMap (implemented). 

Adding user defined volumes (for larger sites ConfigMaps are kind of small) and a tls secret are good future features that don't have to break the current capabilities.

Reply:
> Most of this can be satisfied by the ability to replace `/etc/nginx` with a ConfigMap (implemented).

The line between the API exposed by `values.yaml` and whatever we say the user must define from scratch in a configmap is tricky to get right, but I would prefer for the features I outlined above to be supported directly by the chart since A) they represent extremely common use cases, and B) this makes it easier to test and document supported configurations.

I would be happy to chat in the kubernetes slack about implementation details -- I think there's a lot we could port from the previous (unfortunately closed source) work I've done on this in the past.